tests/test_resolve_command.rs

```diff
@@ -414,3 +410,164 @@ fn test_file_vs_dir() {
 
     "###);
 }
+
+#[test]
+fn test_multiple_conflicts() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    create_commit(
+        &test_env,
+        &repo_path,
+        "base",
+        &[],
+        &[("file1", "first base\n"), ("file2", "second base\n")],
+    );
+    create_commit(
+        &test_env,
+        &repo_path,
+        "a",
+        &["base"],
+        &[("file1", "first a\n"), ("file2", "second a\n")],
+    );
+    create_commit(
+        &test_env,
+        &repo_path,
+        "b",
+        &["base"],
+        &[("file1", "first b\n"), ("file2", "second b\n")],
+    );
+    create_commit(&test_env, &repo_path, "conflict", &["a", "b"], &[]);
+    // Test the setup
+    insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
+        @   conflict
+        |\  
+        o | b
+        | o a
+        |/  
+        o base
+        o 
+        "###);
+    insta::assert_snapshot!(
+    std::fs::read_to_string(repo_path.join("file1")).unwrap()
+        , @r###"
+    <<<<<<<
+    %%%%%%%
+    -first base
+    +first a
+    +++++++
+    first b
+    >>>>>>>
+    "###);
+    insta::assert_snapshot!(
+    std::fs::read_to_string(repo_path.join("file2")).unwrap()
+        , @r###"
+    <<<<<<<
+    %%%%%%%
+    -second base
+    +second a
+    +++++++
+    second b
+    >>>>>>>
+    "###);
+
+    let editor_script = test_env.set_up_fake_editor();
+
+    // Check that we can manually pick which of the conflicts to resolve first
+    std::fs::write(&editor_script, "expect\n\0write\nresolution file2\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["resolve", "file2"]);
+    insta::assert_snapshot!(test_env.jj_cmd_success(&repo_path, &["diff", "-r", "conflict"]), 
```

Comment:
nit: you can drop the `-r conflict` from this call and the following ones (but feel free to keep it if you prefer)

Reply:
You're right, it makes no sense to do `resolve` without `-r conflict` and the `diff` with it.

Which reminds me, I should add a test that tests `resolve -r different_revision`. Will add a TODO for now.